app/common/auth/authorisation_helper.py

```diff
@@ -0,0 +1,82 @@
+from uuid import UUID
+
+from app.common.data.models_user import User
+from app.common.data.types import GRANT_ROLES_MAPPING, RoleEnum
+
+
+class AuthorisationHelper:
+    @staticmethod
+    def has_logged_in(user: User) -> bool:
+        # FIXME: We should have some actual tracking of whether the user has logged in. This could either be a
+        #        field on the model called `last_logged_in_at` or similar, or we could only create entries in the user
+        #        table when the user actually logs in, rather than at invitation-time. Then we could simply trust that
+        #        if a user entry exists, they have definitely logged in.
+        return bool(user.name)
+
+    @staticmethod
+    def is_platform_admin(user: User) -> bool:
+        is_platform_admin = any(
+            role.role == RoleEnum.ADMIN and role.organisation_id is None and role.grant_id is None
+            for role in user.roles
+        )
+        return is_platform_admin
+
+    @staticmethod
+    def is_grant_admin(grant_id: str, user: User) -> bool:
```

Comment:
Please we can make sure that UUIDs are typed as UUIDs everywhere, and that we don't pass them around as strings.

Although we're used to seeing UUIDs as strings, UUIDs are 128-bit values. Their 36-character alphanumeric representation is a stringified version of a UUID, but it is not the same as the UUID itself.

You can get some subtle bugs if you rely on treating them as strings, because things like case-sensitivity come into the equation.

All routes that take in UUIDs should type them properly as UUIDs in the decorator and so you should receive proper UUID objects in the view.

If we can feed this through everywhere it would be great and will prevent us having some strange problem in the future. For example: https://github.com/alphagov/notifications-admin/pull/4822